### DISCUSSION TITLE
I always run into the same error when I start the training process. Would be greatful for any help!

### DISCUSSION THREAD

**Discussion by Justblamejay** [bmaltais/kohya_ss#313] [Q&A]
By now I have already reinstalled stable diffusion, git and python probably 2 times in that the problem lies there.
If anyone could help me I would be really thankful!
Every time I start the training process I gett this error:

===================================BUG REPORT===================================
Welcome to bitsandbytes. For bug reports, please submit your error trace to: https://github.com/TimDettmers/bitsandbytes/issues
For effortless bug reporting copy-paste your error into this form: https://docs.google.com/forms/d/e/1FAIpQLScPB8emS3Thkp66nvqwmjTEgxp8Y9ufuWTzFyr9kJ5AoI47dQ/viewform?usp=sf_link
================================================================================
CUDA SETUP: Loading binary C:\AI\Kohya\kohya_ss\venv\lib\site-packages\bitsandbytes\libbitsandbytes_cuda116.dll...
use 8-bit AdamW optimizer | {}
running training / 学習開始
  num train images * repeats / 学習画像の数×繰り返し回数: 1800
  num reg images / 正則化画像の数: 0
  num batches per epoch / 1epochのバッチ数: 1800
  num epochs / epoch数: 1
  batch size per device / バッチサイズ: 1
  total train batch size (with parallel & distributed & accumulation) / 総バッチサイズ（並列学習、勾配合計含む）: 1
  gradient accumulation steps / 勾配を合計するステップ数 = 1
  total optimization steps / 学習ステップ数: 1800
fatal: not a git repository (or any of the parent directories): .git
steps:   0%|                                                                                  | 0/1800 [00:00<?, ?it/s]epoch 1/1
C:\AI\Kohya\kohya_ss\venv\lib\site-packages\torch\utils\checkpoint.py:25: UserWarning: None of the inputs have requires_grad=True. Gradients will be None
  warnings.warn("None of the inputs have requires_grad=True. Gradients will be None")
Error no kernel image is available for execution on the device at line 167 in file D:\ai\tool\bitsandbytes\csrc\ops.cu
Traceback (most recent call last):
  File "C:\Users\Miedler\AppData\Local\Programs\Python\Python310\lib\runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\Miedler\AppData\Local\Programs\Python\Python310\lib\runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "C:\AI\Kohya\kohya_ss\venv\Scripts\accelerate.exe\__main__.py", line 7, in <module>
  File "C:\AI\Kohya\kohya_ss\venv\lib\site-packages\accelerate\commands\accelerate_cli.py", line 45, in main
    args.func(args)
  File "C:\AI\Kohya\kohya_ss\venv\lib\site-packages\accelerate\commands\launch.py", line 1104, in launch_command
    simple_launcher(args)
  File "C:\AI\Kohya\kohya_ss\venv\lib\site-packages\accelerate\commands\launch.py", line 567, in simple_launcher
    raise subprocess.CalledProcessError(returncode=process.returncode, cmd=cmd)
subprocess.CalledProcessError: Command '['C:\\AI\\Kohya\\kohya_ss\\venv\\Scripts\\python.exe', 'train_network.py', '--pretrained_model_name_or_path=runwayml/stable-diffusion-v1-5', '--train_data_dir=C:/AI/Sophie/Resized/Processed/Sophie Maus LORA/image', '--resolution=512,512', '--output_dir=C:/AI/Sophie/Resized/Processed/Sophie Maus LORA/model', '--logging_dir=C:/AI/Sophie/Resized/Processed/Sophie Maus LORA/log', '--network_alpha=128', '--save_model_as=safetensors', '--network_module=networks.lora', '--text_encoder_lr=5e-5', '--unet_lr=0.0001', '--network_dim=128', '--output_name=Sophie Maus', '--lr_scheduler_num_cycles=1', '--learning_rate=0.0001', '--lr_scheduler=constant', '--train_batch_size=1', '--max_train_steps=1800', '--save_every_n_epochs=1', '--mixed_precision=fp16', '--save_precision=fp16', '--seed=1234', '--caption_extension=.txt', '--cache_latents', '--optimizer_type=AdamW8bit', '--max_data_loader_n_workers=1', '--clip_skip=2', '--bucket_reso_steps=64', '--mem_eff_attn', '--gradient_checkpointing', '--xformers', '--bucket_no_upscale']' returned non-zero exit status 1.

**Comment by bmaltais** [comment]
This error is stumping me... anyu idea what might be causing that on your system?

`fatal: not a git repository (or any of the parent directories): .git`

**Reply by Justblamejay** [reply]
I don't know what was causing it but this I could solve.
However this error line confuses me the most:
`Error no kernel image is available for execution on the device at line 167 in file D:\ai\tool\bitsandbytes\csrc\ops.cu`
I don't even have a directory called ai on my D drive

**Reply by Justblamejay** [reply]
I solved the problem! It was the AdamW8bit. As soon as I changed it to the AdamW one it worked

**Reply by jakesee** [reply]
For those looking for answer, "AdamW" is under Dreambooth Lora > Training Parameters > Optimizer > AdamW
My machine is nvidia GPU 6GB ram only.
I also upgraded from 3.10.6 to 3.10.9 before switching to AdamW.
Not sure whether only need to change the optimizer or need to also upgrade python.
After that, I overcome the error "Error no kernel image is available for execution on the device at line 167 in file D:\ai\tool\bitsandbytes\csrc\ops.cu"

**Comment by pandamontok** [comment]
probably you need python 3.10.9 as suggested.
I used 3.10.6 before, cos I learned automatic1111 first. 
But then I changed it. simillar error gone. 
I can use AdamW8bit and everything.
